File: typescript/api/src/base/gameObject.ts
import {AbstractGameEnity} from "./abstractGameEnity";
import {PlayerSession} from "../playerSession";

/**
* Abstract class that represents an object that can be in a room
* All other object should extend this class
*
*
*/
export abstract class GameObject extends AbstractGameEnity {

    public smell(_playerSession: PlayerSession): string {
        return "How weird";
    }

    public move(_playerSession: PlayerSession): string {
        return "You idiot!";
    }

    public open(_playerSession: PlayerSession): string {
        return "You can't";
    }

    public talkTo(_playerSession: PlayerSession): string {
        return "no response";
    }

}

File: java/api/src/main/java/game/engine/java/api/base/GameObject.java
package game.engine.java.api.base;


import game.engine.java.api.PlayerSession;

/**
 * Abstract class that represents any form of object in the game
 * Abstract classes Room and GameObject implement this class
 */
public abstract class GameObject extends AbstractGameEntity {

    public GameObject(String id) {
        super(id);
    }

    public String smell(PlayerSession playerSession) {
        return "How weird";
    }

    public String move(PlayerSession playerSession) {
        return "You idiot!";
    }

    public String open(PlayerSession playerSession) {
        return "You can't";
    }

    public String talkTo(PlayerSession playerSession) {
        return "no response";
    }

}
